client/my-sites/plugins/marketplace/components/simulated-progressbar/index.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { ProgressBar } from '@automattic/components';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import styled from '@emotion/styled';
+
+const Container = styled.div`
+	text-align: center;
+`;
+const StyledProgressBar = styled( ProgressBar )`
+	margin: 20px 0px;
+`;
+
+const Title = styled.h1`
+	font-size: 2rem;
+`;
+
+export function resolveStep(
+	steps: TranslateResult[],
+	currentPercentage: number
+): TranslateResult {
+	const totalSteps = steps.length;
+	const perStepPercentage = 100 / totalSteps;
+	const quotient = Math.floor( currentPercentage / perStepPercentage );
+
+	if ( currentPercentage <= perStepPercentage ) {
+		return steps[ 0 ];
+	} else if ( currentPercentage >= perStepPercentage * totalSteps ) {
+		return steps[ steps.length - 1 ];
+	}
+	return steps[ quotient ];
+}
+
+export default function SimulatedProgressbar( {
+	steps,
+}: {
+	steps: TranslateResult[];
+} ): JSX.Element {
+	const translate = useTranslate();
+	const [ currentStep, setCurrentStep ] = useState( steps[ 0 ] );
+	const [ simulatedProgressPercentage, setSimulatedProgressPercentage ] = useState( 1 );
+
+	const SIMULATION_REFRESH_INTERVAL = 2000;
+	const INCREMENTED_PERCENTAGE_SIZE_ON_STEP = 6;
+	const MAX_PERCENTAGE_SIMULATED = 100 - INCREMENTED_PERCENTAGE_SIZE_ON_STEP * 2;
```

Comment:
Would it be a problem to add a few empty lines here and there for consistency?

Reply:
Hey Veselin I did include some empty line spacing for clarity, can you share some places where would like to see a bit more? 🤔